buildpacktoml.go

```diff
@@ -254,6 +262,20 @@ func (b *BuildpackTOML) readOutputFiles(bpLayersDir, bpPlanPath string, bpPlanIn
 	return br, nil
 }
 
+// check that there are not multiple default processes in a buildpack with different process types
+func validateProcesses(processes []launch.Process) error {
+	defaultType := ""
```

Comment:
Does this validation block need to be in a version guard for either platform or bp API?

Reply:
I had the same question myself and I'm not sure...

Reply:
mmmm...this validation is relevant only for `buildpackAPI >= 0.6`, but since for older versions this validation will never fail (because we introduced the `default` field only in `buildpackAPI 0.6`), we thought it's ok to not check it.
What do you think?

Reply:
That sounds good to me!

Reply:
Related [Slack conversation](https://buildpacks.slack.com/archives/CSFE49VGR/p1612291496030300)

Reply:
It also relates to [this comment](https://github.com/buildpacks/lifecycle/pull/520#discussion_r569081128).

Reply:
To summarize the various conversations, I think when a buildpack declaring an older buildpack api tries to implement a newer buildpack api feature, we should warn and ignore the configuration. So in this case if a buildpack on 0.5 writes `default = true` to launch.toml, we should warn and set `default = false`.

Reply:
Please see https://github.com/buildpacks/lifecycle/pull/520/commits/8ef28ee8fc57b4ca50e83bd20938479fbcaf60e9.